plugin/lighthouse/handler.go

```diff
@@ -61,26 +65,83 @@ func (lh *Lighthouse) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns
 		return lh.nextOrFailure(state.Name(), ctx, w, r, dns.RcodeNameError, "Only services supported")
 	}
 
+	return lh.getDNSRecord(zone, state, ctx, w, r, pReq)
+}
+
+func (lh *Lighthouse) createARecords(ips []string, state request.Request) []dns.RR {
+	records := make([]dns.RR, 0)
+
+	for _, ip := range ips {
+		record := &dns.A{Hdr: dns.RR_Header{Name: state.QName(), Rrtype: dns.TypeA, Class: state.QClass(), Ttl: lh.ttl}, A: net.ParseIP(ip).To4()}
+		log.Debugf("rr is %v", record)
+		records = append(records, record)
+	}
+
+	return records
+}
+
+func (lh *Lighthouse) createSRVRecords(record *serviceimport.DNSRecord, state request.Request, pReq recordRequest, zone string) []dns.RR {
+	reqPorts := make([]mcsv1a1.ServicePort, 0)
+	records := make([]dns.RR, 0)
+	portReqKey := pReq.port + pReq.protocol
+	if portReqKey == "" {
+		reqPorts = record.Port
+	}
+
+	for _, port := range record.Port {
+		portKey := strings.ToLower(port.Name) + strings.ToLower(string(port.Protocol))
+		log.Debugf("The current port Name %q SRV require port Name %q", portKey, portReqKey)
+
+		if portKey == portReqKey {
+			reqPorts = append(reqPorts, port)
+		}
+	}
```

Comment:
If no port is requested then we're unnecessarily looping on `record.Port`.  Also what if` pReq.port` is empty but `pReq.protocol` isn't?  How about: 

```suggestion
	if pReq.port == "" {
		reqPorts = record.Port
	} else {
	    log.Debugf("Requested port %q, protocol %q for SRV", pReq.port, pReq.protocol)
	    for _, port := range record.Port {
		name := strings.ToLower(port.Name)
		protocol := strings.ToLower(string(port.Protocol))
		
		log.Debugf("Checking port %q, protocol %q", name, protocol)

		if name == pReq.port && protocol ==  pReq.protocol {
			reqPorts = append(reqPorts, port)
		}
	    }
	}
```

Reply:
Done.